psake.psm1

```diff
@@ -621,7 +621,23 @@ function ConfigureBuildEnvironment {
     $frameworkDirs = @()
     if ($buildToolsVersions -ne $null) {
         foreach($ver in $buildToolsVersions) {
-            if (Test-Path "HKLM:\SOFTWARE\Microsoft\MSBuild\ToolsVersions\$ver") {
+            if ($ver -eq "15.0") {
+                $vs2017paths = @(
+                    "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise",

```

Comment:
Please take a look at how it was done in https://github.com/nightroman/Invoke-Build/blob/master/Resolve-MSBuild.ps1

So far nobody reports issues.